renderdoc/driver/vulkan/vk_debug.cpp

```diff
@@ -1393,6 +1395,22 @@ void VulkanReplay::CreateResources()
       });
 
   CREATE_OBJECT(m_MeshFetchDescSet, m_General.DescriptorPool, m_MeshFetchDescSetLayout);
+
+  GPA_vkContextOpenInfo context = {Unwrap(m_pDriver->GetInstance()),
+                                   Unwrap(m_pDriver->GetPhysDev()), Unwrap(m_pDriver->GetDev())};
+
+#if ENABLED(RDOC_WIN32) || ENABLED(RDOC_LINUX)
```

Comment:
I don't like how many places have to be protected by this `#if` here, having platform preprocessor conditionals around the code quickly reduces readability. I realise you don't support the counters except on windows and linux, but that should be handled by the library failing to load and the initialisation failing but otherwise executing the same single code-path.

I assume there are compile errors on android/OS X. Are they fixable? If so that would be ideal, but if not I'd rather add a `amd_counters_stub.cpp` for those platforms and define a stub implementation that just always fails on those other platforms.